### PR TITLE
[CWS] make use of statsd client interface

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This will help to write test using statsd mock.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/probe/discarder_monitor.go [hunk 1]
```diff
@@ -28,7 +27,7 @@ type DiscarderStats struct {
 
 // DiscarderMonitor defines a discarder monitor
 type DiscarderMonitor struct {
-	client            *statsd.Client
+	probe             *Probe
```
Comment (paulcacheux) [inline pkg/security/probe/discarder_monitor.go:30]: Do we really need the whole probe, why use `client statsd.ClientInterface` ?

**Review by paulcacheux — COMMENTED** [review]
This feels a bit risky for some reasons, do we have tests to check that metrics are still being propagated correctly ?

File: pkg/security/probe/discarder_monitor.go (hunk 1 above)
Comment (safchain) [reply pkg/security/probe/discarder_monitor.go:30]: Will use the interface instead of probe

**Comment by safchain** [comment]
I don't think it is risky, I'm just using an interface, the client is still the same. The goal of this PR is indeed to add test for metrics that is why I replaced the client by its interface.

**Review by brycekahle — APPROVED** [review]
👍  for agent-network owned files

**Review by paulcacheux — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**